### PR TITLE
Crawler filters aren t merged properly [sc-25950]

### PR DESCRIPTION
**Pull request by alyiwang**
### 🤔 Why?

The previous merge filter function will take two database filters and merge together with overwriting, i.e. if two filters has same key, then the second value will overwrite the first value. What we want instead is to merge the two values (schema filters) together. 

### 🤓 What?

- change the merge logic to actually merge the database filter values instead of overwriting. However, the schema filter level, we still do merge with overwriting.

### 🧪 Tested?

unit tests

### ☑️ Checks

- [x] My PR contains actual code changes, and I have updated the version number in `pyproject.toml`.

### CODE REVIEW

**Review by mars-lan — COMMENTED** [review]
Also need to merge the table patterns?

**Comment by alyiwang** [comment]
> Also need to merge the table patterns?

updated

File: tests/common/test_filter.py [hunk 1]
```diff
@@ -207,6 +207,64 @@ def test_merge():
...
+            "*": {
+                "foo": {"f1", "f2", "f3"},
+                "bar": {"b1"},
+            },
+        }
+    )
+
+    f1 = DatasetFilter(
+        excludes={
+            "*": {"foo": {"f1", "f2"}, "bar": None},
```
Comment (mars-lan) [inline tests/common/test_filter.py:255]: Consider replacing `*` with an actual schema name to show that it's not just limited to `*`?

**Review by mars-lan — COMMENTED** [review]
Otherwise LGTM

File: tests/common/test_filter.py (hunk 1 above)
Comment (alyiwang) [reply tests/common/test_filter.py]: updated

**Review by mars-lan — APPROVED** [review]

**Pull request merged**